sdk/core/core-amqp/src/errors.ts

```diff
@@ -481,14 +481,24 @@ export const retryableErrors: string[] = [
   "ServerBusyError",
   "ServiceUnavailableError",
   "OperationCancelledError",
+
+  // OperationTimeoutError occurs when the service fails to respond within a given timeframe.
+  // Since reasons for such failures can be transient, this is treated as a retryable error.
   "OperationTimeoutError",
+
   "SenderBusyError",
   "MessagingError",
   "DetachForcedError",
   "ConnectionForcedError",
-  "TransferLimitExceededError"
+  "TransferLimitExceededError",
+
+  // InsufficientCreditError occurs when the number of credits available on Rhea link is insufficient.
+  // Since reasons for such shortage can be transient such as for pending delivery of messages, this is treated as a retryable error.
+  "InsufficientCreditError"
 ];
 
+export const nonRetryableErrors: string[] = ["AbortError"];
```

Comment:
We can create an array like this when we encounter more custom non retryable errors. For now, let's use `AbortError` directly in `getCustomError`